Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/Driver/GoHomeRequest.hs

```diff
@@ -0,0 +1,143 @@
+  Hedis.setExp ghKey (templateGoHomeData (Just DDGR.ACTIVE) ghInfo.cnt (Just vTill) (Just guId) False (Just ghInfo.goHomeReferenceTime) currTime) expTime
+  where
+    buildDriverGoHomeRequest guId driverHomeLocation = do
+      let id = guId
+      now <- getCurrentTime
+      return $
+        DDGR.DriverGoHomeRequest
+          { lat = driverHomeLocation.lat,
+            lon = driverHomeLocation.lon,
+            status = DDGR.ACTIVE,
+            numCancellation = 0,
+            createdAt = now,
+            updatedAt = now,
+            ..
+          }
+
+deactivateDriverGoHomeRequest :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Merchant -> Id DP.Driver -> DDGR.DriverGoHomeRequestStatus -> CachedGoHomeRequest -> m ()
+deactivateDriverGoHomeRequest merchantId driverId stat ghInfo = do
+  currTime <- getLocalCurrentTime =<< ((CQTC.findByMerchantId merchantId >>= fromMaybeM (InternalError "Transporter config for timezone not found")) <&> (.timeDiffFromUtc))
+  let ghKey = makeGoHomeReqKey driverId
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  driverGoHomeReqId <- fromMaybeM (InternalError "Could not Find DriverGoHomeRequestId") ghInfo.driverGoHomeRequestId
+  case stat of
+    SUCCESS -> do
+      QDGR.finishWithStatus driverGoHomeReqId DDGR.SUCCESS
+      Hedis.setExp ghKey (templateGoHomeData Nothing (ghInfo.cnt - 1) Nothing Nothing False (Just ghInfo.goHomeReferenceTime) currTime) expTime
+    FAILED -> do
+      logDebug "Here:- inside deactivate driver go home request"
+      QDGR.finishWithStatus driverGoHomeReqId DDGR.FAILED
+      Hedis.setExp ghKey (templateGoHomeData Nothing ghInfo.cnt Nothing Nothing False (Just ghInfo.goHomeReferenceTime) currTime) expTime
+
+resetDriverGoHomeRequest :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Merchant -> Id DP.Driver -> GoHomeConfig -> CachedGoHomeRequest -> m ()
+resetDriverGoHomeRequest merchantId driverId goHomeConfig ghInfo = do
+  currTime <- getLocalCurrentTime =<< ((CQTC.findByMerchantId merchantId >>= fromMaybeM (InternalError "Transporter config for timezone not found")) <&> (.timeDiffFromUtc))
+  let ghKey = makeGoHomeReqKey driverId
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  Hedis.setExp ghKey (templateGoHomeData ghInfo.status ghInfo.cnt (Just $ addUTCTime (fromIntegral goHomeConfig.activeTime) currTime) ghInfo.driverGoHomeRequestId False (Just ghInfo.goHomeReferenceTime) currTime) expTime
+
+increaseDriverGoHomeRequestCount :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Merchant -> Id DP.Driver -> m ()
+increaseDriverGoHomeRequestCount merchantId driverId = do
+  currTime <- getLocalCurrentTime =<< ((CQTC.findByMerchantId merchantId >>= fromMaybeM (InternalError "Transporter config for timezone not found")) <&> (.timeDiffFromUtc))
+  let ghKey = makeGoHomeReqKey driverId
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  ghInfo <- getDriverGoHomeRequestInfo driverId merchantId Nothing
```

Comment:
what if this API gets called before driver has enabled go home for today ? this will give you dummy template with no gohomeReqId right ?

Reply:
yes. But not dummy data as count will be there and reference time to keep track of count will be there.